### ISSUE TITLE
Stat curse from artifact not wearing off

### ISSUE THREAD

**Issue by LScott1** [CleverRaven/Cataclysm-DDA#47601]
# Describe the bug
The effect of a cursed artifact won't wear off. The curse drops all stats by 10, might be more to it. I've dropped all my artifacts away from my base in a nearby house, and have waited several game days though the curse is still in effect. Upon reloading the save, stats will be normal for a turn before the effect returns. I did originally save and quit while I had the curse originally, which may have messed up the timer for it. My hidden health stat is tanked but I don't see why that would matter for curses, and if it's supposed to take over 3 days to wear off in this case I'm surprised. I can't find anything specific online about this curse but it started around when I looted a couple new artifacts.

# Steps To Reproduce
1. Pick up an artifact with the -10 stat curse. Effect activates. 
2. Discard artifact.
3. Save and reload while curse is in effect (maybe not required) 
4. Stats will be normal, wait a turn
5. Stats will be reduced by 10 again, and no amount of waiting will remove the curse (waited over 3 days)

# Expected behavior
The curse should wear off over time.

# Screenshots
Upon reloading: 
![image](https://user-images.githubusercontent.com/54735763/108467962-04e70380-7254-11eb-89fc-714adcc90e59.png)
A few turns later, curse is back in effect (and to stay):
![image](https://user-images.githubusercontent.com/54735763/108468005-15977980-7254-11eb-9775-ff2f7f2b805e.png)
Inventory:
![image](https://user-images.githubusercontent.com/54735763/108469217-db2edc00-7255-11eb-8d22-f39f74c86896.png)
![image](https://user-images.githubusercontent.com/54735763/108470857-935d8400-7258-11eb-886f-20edda54debb.png)

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E-3 [64-bit]
- Graphics Version: Tiles
- Game Language: System language [] (English)
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Monsters [No_Fungi],
    Faster Stamina Regen [stamina_regen_buff]
]

Save file:
[Reedsburg_copy.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6011582/Reedsburg_copy.zip)

**Comment by Fosheze** [comment]
I'll see if I can take a look at the save after work. But if you can send me the artifact names before then I can look at what all of them are supposed to do. It doesn't sound like it bit it is always possible that this is intended behavior. From what I understand, artifacts are currently just kind of a hold over waiting to be phased out or revamped so I wouldn't be surprised if something broke them and it went unnoticed.

**Comment by actual-nh** [comment]
> I'll see if I can take a look at the save after work. But if you can send me the artifact names before then I can look at what all of them are supposed to do.

I'm not sure if the artifact names have any correspondence with the side effects; IIRC, there is some random assignment.

**Comment by Fosheze** [comment]
> > I'll see if I can take a look at the save after work. But if you can send me the artifact names before then I can look at what all of them are supposed to do.
> 
> I'm not sure if the artifact names have any correspondence with the side effects; IIRC, there is some random assignment.

It looks like you're right. I'm not sure where I saw that the names corresponded with the effects. I also just looked through all of the possible effect and there is nothing that reduces all stats by 10 and nothing that reduces stats in an AOE that I can see. The closest I found is "AEP_ALL_DOWN" but that only reduces all stats by 2 and requires the artifact to be wielded. I'm also seeing some of the proc gen stuff that can reduce an individual stat by up to 3. I suppose I could still take a look at the actual save files quick.

**Comment by Fosheze** [comment]
@LScott1 I just noticed that you only attached the artifacts.gsav. It would help if you attached the whole world file so that we can see what all of the actual effects applied to your character are.

**Comment by LScott1** [comment]
Alright, added to the original post.

**Comment by Fosheze** [comment]
Well I found out what's causing it.
`"effects": { "evil": { "12": { "eff_type": "evil", "duration": 319142, "bp": 12, "permanent": false, "intensity": 1, "start_turn": 17307235 }`
Why the duration is so long I have no idea. It also looks like that is supposed to be a debug only effect. I'm not seeing anywhere where it has a chance to be applied to an item on it's own.
https://github.com/CleverRaven/Cataclysm-DDA/blob/703dd607ee7d6229fc7229a76da2aa2a93dd16d9/data/json/artifact/legacy_artifact_passive.json#L81-L93
https://github.com/CleverRaven/Cataclysm-DDA/blob/5cc608ccea74ba783c2f1569fafdbc2b9caf0021/src/player_hardcoded_effects.cpp#L588-L620

I believe I fixed your save but I can't test it to be sure. If you drop [This File.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6011724/This.File.zip) into your save folder and choose yes to overwrite it should remove the evil effect from your character. It also looks like the the "Bladed Harp of Shattered Death" is what was causing the effect so as long as you don't pick it up again you should be fine.

**Comment by LScott1** [comment]
Yeah that fixed it, I really appreciate it. And that's really bizarre.

**Comment by LScott1** [comment]
I wasn't holding them very long before I dropped them back in containers and all, got maybe a couple messages hinting at the evil modifier but after those made sure to keep them off of me as much as possible, and never rested with them on me if it helps to know.

**Comment by Fosheze** [comment]
> I wasn't holding them very long before I dropped them back in containers and all, got maybe a couple messages hinting at the evil modifier but after those made sure to keep them off of me as much as possible, and never rested with them on me if it helps to know.

All you need to do is pick an artifact up for a second and it's passive effects immediately trigger. I'm not sure if moving it from tile to tile with advanced inventory triggers the effect or not.

**Comment by LScott1** [comment]
I'm sorry, just should I close this yet or mark it differently since it's still probably a bug? Maybe just mark as fixing?

**Comment by LScott1** [comment]
Well I suppose if it is a bug, you said artifacts were going to be revamped soon anyways.

**Comment by Fosheze** [comment]
> I'm sorry, just should I close this yet or mark it differently since it's still probably a bug?

It's probably a good idea to leave it open because there is still a bug here. You still should not have had the effect lasting that long and unless you debug spawned the item it shouldn't have had that effect in the first place.
